server/src/main/java/org/elasticsearch/cluster/ClusterModule.java

```diff
@@ -172,6 +174,11 @@ public static List<Entry> getNamedWriteables() {
 
         // Task Status (not Diffable)
         entries.add(new Entry(Task.Status.class, PersistentTasksNodeService.Status.NAME, PersistentTasksNodeService.Status::new));
+
+        // Health node selector task is always present in the cluster state
+        if (HealthNodeSelector.isEnabled()) {
```

Comment:
What are the guarantees we give for a feature flag? If someone run with this feature flag enabled, they are going to create the persistent task, if then they decide to disable the feature flag, then master would fail because they will not know how to parse the `HealthNodeSelectorTaskParams`, right? 

Should we not put this behind the feature flag, so even with the feature disabled, if the master encounters this task it will not crash?

Reply:
Good question. Not sure. My thinking would be that this feature flag is to hide an experimental feature so it shouldn't be enabled in production clusters, and therefore we don't really need to support switching it off later. But on the other hand if someone does accidentally enable this in their production cluster then it would be pretty bad to have no way to switch it off again.

Reply:
I am assuming that only adding the writable is not too much overhead and not that many things could go wrong from this... For this reason, I am leaning towards not putting this behind the flag.